include/forcing/ForcingsEngineDataProvider.hpp

```diff
@@ -0,0 +1,213 @@
+#pragma once
+
+#include <cmath>
+#include <chrono>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "DataProvider.hpp"
+#include "bmi/Bmi_Py_Adapter.hpp"
+
+namespace data_access {
+
+//! Python module name for NextGen Forcings Engine
+static constexpr auto forcings_engine_python_module = "NextGen_Forcings_Engine";
+
+//! Python classname for NextGen Forcings Engine BMI class
+static constexpr auto forcings_engine_python_class  = "NWMv3_Forcing_Engine_BMI_model";
+
+//! Full Python classpath for Forcings Engine BMI class
+static constexpr auto forcings_engine_python_classpath = "NextGen_Forcings_Engine.NWMv3_Forcing_Engine_BMI_model";
+
+//! Default time format used for parsing timestamp strings
+static constexpr auto default_time_format = "%Y-%m-%d %H:%M:%S";
+
+namespace detail {
+
+//! Parse time string from format.
+//! Utility function for ForcingsEngineLumpedDataProvider constructor.
+time_t parse_time(const std::string& time, const std::string& fmt);
+
+//! Check that requirements for running the forcings engine
```

Comment:
Since this is probably a fatal error only this function is fine I would still want a check form of the function but it is not important